lib/rewardable/Interface.cpp

```diff
@@ -287,6 +287,15 @@ void Rewardable::Interface::grantAllRewardsWithMessage(const CGHeroInstance * co
 
 void Rewardable::Interface::doHeroVisit(const CGHeroInstance *h) const
 {
+	auto instance = dynamic_cast<const CGObjectInstance *>(this);
+	if(!instance)
+	{
+		logGlobal->error("MapObject at tile %s is not a valid CObjectInstance", 
+			getObject()->visitablePos().toString());
+		// throw ?
+		return;
+	}
+
```

Comment:
What about town buildings?
They are IObjectInterface, but they are not CGObjectInstance